Added Shuffle

grimborg opened the pull request:
I've added in-place Shuffle to slices similar to how Underscore does it.

grimborg committed:
Added shuffle

clipperhouse commented:
Looks good! It’s idiomatic in the slice typewriter to return a new slice (instead of in-place). Make that adjustment?

And, if you’re in the mood you can update the docs, they are on the gh-pages branch here: https://github.com/clipperhouse/gen/tree/gh-pages ([see also](https://pages.github.com/))

Thanks!

grimborg commented:
Thanks for the feedback, Matt.

I've changed it so that the slice is copied before shuffling (as Sort does). I've also updated the docs (https://github.com/clipperhouse/gen/pull/75).

clipperhouse commented:
Thanks!
